src/components/composer/content_editable_helper.ts

```diff
@@ -105,16 +105,25 @@ export class ContentEditableHelper {
   getCurrentSelection() {
     let { startElement, endElement, startSelectionOffset, endSelectionOffset } =
       this.getStartAndEndSelection();
-    let startSizeBefore = this.findSizeBeforeElement(startElement!);
-    let endSizeBefore = this.findSizeBeforeElement(endElement!);
+
+    let startSizeBefore = this.findSelectionIndex(startElement!, startSelectionOffset);
+    let endSizeBefore = this.findSelectionIndex(endElement!, endSelectionOffset);
 
     return {
-      start: startSizeBefore + startSelectionOffset,
-      end: endSizeBefore + endSelectionOffset,
+      start: startSizeBefore,
+      end: endSizeBefore,
     };
   }
 
-  private findSizeBeforeElement(nodeToFind: Node): number {
+  /**
+   * FComputes the text 'index' inside this.el based on the currently selected node and its offset.
+   * the offset might be inside a text node or inside a child node. In the latter case, we have to
+   * take account of all previous nodes contents. See the MDN documentation for more details.
```

Comment:
If we only read the docstring, it's a bit unclear that if(typeof node === TextNode), offset unit = number of characters, else offset unit = number of child nodes

Reply:
well that's what the hyperlinks are for, i don't feel like it's relevant to explain things in full details here